squape/object_tree.py

```diff
@@ -269,3 +271,85 @@ def _get_object_reference(object_or_name: any) -> any:
         # Symbolic name
         return squish.waitForObjectExists(object_or_name)
     return object_or_name
+
+
+@objectNotFoundDebugging(False)
+def _wait_for_any_object(
+    lookup_function, object_names: list, timeout: int, retry_delay: float
+):
+    lookup_errors = set()
+    start_time = time.time()
+
+    while True:
+        for obj_name in object_names:
+            try:
+                return lookup_function(obj_name, 0)
+
+            except LookupError as e:
+                lookup_errors.add(str(e))
+
+        elapsed_time = time.time() - start_time
+        if elapsed_time > timeout:
+            break
+
+        squish.snooze(retry_delay)
+
+    raise LookupError(f"Objects not found: {lookup_errors}")
```

Comment:
It seems that you may end up with tons of lookup errors. Each object may cause different lookup errors over time, if you multiply it by the number of objects things can get crazy :P
Maybe we should report just one but mentioning all the objects that were provided in `object_names`?
I know that different objects may not be accessible due to a different reasons. How about adding DEBUG messages printing details of the **last** lookup error for each object?

Reply:
I have added debug messages with [397aed9](https://github.com/CyberAlpaca/squish-api-python-extension/pull/52/commits/397aed9392694d2053ced3f00d1868f46becce99). A debug message is provided for each lookup error try.

I don't agree with the statement that if no object is found, you will end up with a lot of lookup errors. I think that in most cases, you will have a single error message per object_name from a list.

Reply:
Changed with [7a137ac](https://github.com/CyberAlpaca/squish-api-python-extension/pull/52/commits/7a137ac0cfed39855d2e7c0a7cc0d4b6bd28e7c9).
- Debug message with lookup error will be printed for every lookup error (for every try).
- Final error message will not contain lookup errors:
` raise LookupError(f"Objects {object_names} not found")`